hello/test/test.c

```diff
@@ -0,0 +1,26 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <assert.h>
+#include "../src/hello.h"
+
+int
+main(void) {
+
+  char *foo = hello("foo");
+  char expected[12] = "Hello: foo\n";
+
+  if (strcmp(foo, expected)) {
+    printf("\n");
+    printf("  expected: '%s'\n", foo);
+    printf("    actual: '%s'\n", expected);
+    printf("\n");
+    free(foo);
+    exit(EXIT_FAILURE);
```

Comment:
You use exit() because this is test code, aren't you?
Maybe you can use return such as situation.